libraries/AP_BattMonitor/AP_BattMonitor.cpp

```diff
@@ -175,11 +175,17 @@ AP_BattMonitor::init()
 #if CONFIG_HAL_BOARD == HAL_BOARD_PX4
                 drivers[instance] = new AP_BattMonitor_SMBus_PX4(*this, instance, state[instance]);
 #else
-                drivers[instance] = new AP_BattMonitor_SMBus_I2C(*this, instance, state[instance],
+                drivers[instance] = new AP_BattMonitor_SMBus_Solo(*this, instance, state[instance],
                                                                  hal.i2c_mgr->get_device(BATTMONITOR_SBUS_I2C_BUS, BATTMONITOR_SMBUS_I2C_ADDR));
 #endif
                 _num_instances++;
                 break;
+            case BattMonitor_TYPE_MAXCELL:
```

Comment:
I think that it is unnecessary if it can distinguish by the value of I2C registers.